src/connectors/ledger.js

```diff
@@ -0,0 +1,104 @@
+import TransportU2F from "@ledgerhq/hw-transport-u2f"
+import AppEth from '@ledgerhq/hw-app-eth'
+import { ConnectorUpdate } from '@web3-react/types'
+import { AbstractConnector } from '@web3-react/abstract-connector'
+import Web3ProviderEngine from 'web3-provider-engine'
+import { LedgerSubprovider } from './ledger_subprovider'
+import CacheSubprovider from 'web3-provider-engine/subproviders/cache.js'
+import { RPCSubprovider } from '@0x/subproviders/lib/src/subproviders/rpc_subprovider' // https://github.com/0xProject/0x-monorepo/issues/1400
+import WebsocketSubprovider from 'web3-provider-engine/subproviders/websocket'
+
+/**
+ * An implementation of a LedgerConnector for web3-react, based on the original
+ * `@web3-react/ledger-connector`.
+ * 
+ * Some differences:
+ * 
+ * 1. The original doesn't expose the LedgerJS client API. 
+ *    We will probably want access to this in future, eg. signing BTC transactions
+ * 
+ * 2. The original doesn't work with event subscriptions, as it assumes a HTTP RPC 
+ *    endpoint. Event subscriptions use `eth_subscribe`, which Ganache does not 
+ *    support out-of-the-box. Assuming a Websocket provider is simpler for our case.
+ */
+export class LedgerConnector extends AbstractConnector {
+  constructor({
+    chainId,
+    url,
+    pollingInterval,
+    requestTimeoutMs,
+    accountFetchingConfigs,
+    baseDerivationPath
+  }) {
+    super({ supportedChainIds: [chainId] })
+
+    this.chainId = chainId
+    this.url = url
+    this.pollingInterval = pollingInterval
+    this.requestTimeoutMs = requestTimeoutMs
+    this.accountFetchingConfigs = accountFetchingConfigs
+    this.baseDerivationPath = baseDerivationPath
```

Comment:
`baseDerivationPath` in `LedgerSubprovider` from `0x` is set to `44'/60'/0'`(ledger legacy). Do we want support for ledger live(`44'/60'/0'/0`)?

Reply:
Addressed in #206 with d6e2cfe